### PR TITLE
multi: use getTransaction grpc for confirmCoinbases. 

### PR DESCRIPTION
**Pull request by dnldd**
This reworks coinbase confirmation function to uses the getTransaction grpc to fetch transaction details and confirm them. The rescan begin height is now a coinbase maturity length of blocks from the lowest reported block with a tx having inaccurate confirmation information. Tests have been updated.

### CODE REVIEW

File: pool/paymentmgr.go
Comment (jholdstock) [inline pool/paymentmgr.go:969]: If this PR is merged, we obviously need to remove all this commented code first (and any other code which is no longer required).

Comment (jholdstock) [reply pool/paymentmgr.go:969]: Also need to remove `v2` from the name `confirmCoinbasesV2`

File: pool/paymentmgr_test.go
Comment (jholdstock) [inline pool/paymentmgr_test.go:649]: Required before merging

Comment (jholdstock) [reply pool/paymentmgr_test.go:649]: Also ensure tests for v1 are removed.

File: pool/paymentmgr.go
Comment (jholdstock) [inline pool/paymentmgr.go:647]: Dont need this logging if `len(txHashes) == 0`

**Review by jholdstock — COMMENTED** [review]
Added a few comments, mostly for myself to remember I need to re-review some of the points later.

---

The reason we originally used `TxConfNotifications` to track when coinbases become spendable was because of a race condition explained here in #222.

The old `confirmCoinbases` blocked until all of the coinbases were spendable. I don't think the new one is blocked, so it seems like it will be prone to the same race condition.

File: pool/paymentmgr.go
Comment (dnldd) [reply pool/paymentmgr.go:969]: yeah this isn't expected to be merged any time soon, the plan is to test this extensively to confirm it solves the inaccurate confirmation issue at least after a rescan. With that done I can remove the functions no longer needed, update the tests and get it ready for merging.

File: pool/paymentmgr_test.go
Comment (dnldd) [reply pool/paymentmgr_test.go:649]: noted.

**Comment by dnldd** [comment]
I don't think the issue described in #222 is going to be a problem here since `gettxout` was informing us of what the node knew about the provided tx output not factoring in the wallet's state concerning that output (the wallet could be updated late for example). The current solution being used in `confirmCoinbasesV2` is calling the wallet's `gettransaction` grpc for the transactions of interest. If the wallet doesn't know about them yet it will have two more blocks to do so before a rescan is performed. The disparity we had before concerning the information the wallet and node had at a given time on a tx shouldn't be an issue now since we ask the wallet to confirm txs it will need to create the payout transaction directly, however we'll still have to confirm `gettransaction` works as expected and the expanded rescan range is adequate in mitigating the current tx confirmation issues being had with recieving tx confirmation notifications.

**Comment by jholdstock** [comment]
Understood. Will deploy onto my test site and monitor for a few days.

File: pool/hub.go
Comment (jholdstock) [inline pool/hub.go:261]: Cancel is not used anywhere. Should either be used or removed.

File: pool/paymentmgr.go
Comment (jholdstock) [inline pool/paymentmgr.go:486]: I think this comment needs updating/expanding. Please explain what confirmCoinbases is doing, how it is doing it, and why it is necessary.

File: pool/paymentmgr.go
Comment (jholdstock) [inline pool/paymentmgr.go:834]: This `if` seems unnecessary. How can an error occur if there are zero `inputTxHashes`?

**Review by jholdstock — CHANGES_REQUESTED** [review]

File: pool/hub.go
Comment (dnldd) [reply pool/hub.go:261]: Noted, removing.

File: pool/paymentmgr.go
Comment (dnldd) [reply pool/paymentmgr.go:486]: Sure thing, updated.

File: pool/paymentmgr.go
Comment (dnldd) [reply pool/paymentmgr.go:834]: Yeah, that's no needed. Updating.

**Review by jholdstock — APPROVED** [review]

**Pull request merged**
